src/main/java/org/codehaus/mojo/versions/SetMojo.java

```diff
@@ -494,5 +518,26 @@ protected synchronized void update( ModifiedPomXMLEventReader pom )
         }
         log.clearContext();
     }
+    
+    /**
+     * Returns true if the current project is located at the Execution Root Directory (where mvn was launched)
+     * @return
+     */
+    protected boolean isThisTheExecutionRoot()
+    {
+        ContextualLog log = new DelegatingContextualLog( getLog() );
+        log.debug("Root Folder:" + session.getExecutionRootDirectory());
+        log.debug("Current Folder:"+ basedir );
+        boolean result = session.getExecutionRootDirectory().equalsIgnoreCase(basedir.toString());
+        if (result)
+        {
+            log.debug( "This is the execution root." );
+        }
+        else
+        {
+            log.debug( "This is NOT the execution root." );
+        }
```

Comment:
Another option would be to use ` log.debug("This is the execution root? " + result)` to replace the lines 532 to 539.